cmd/build-push-image/main.go

```diff
@@ -194,13 +200,63 @@ func main() {
 	}
 }
 
+func addNexusBuildArgs(args []string, opts options) (ret_args []string) {
+	// log.Printf("nexusURL: %s", opts.nexusURL)
+	if strings.TrimSpace(opts.nexusURL) != "" {
+		nexusUrl, err := url.Parse(opts.nexusURL)
+		if err != nil {
+			log.Printf("Skip adding nexus build args: could not parse nexus url (%s): %s", opts.nexusURL, err)
+			return args
+		}
+		if nexusUrl.Host == "" {
+			log.Printf("Skip adding nexus build args: could not get host in nexus url (%s)", opts.nexusURL)
+			return args
+		}
+		if opts.nexusUsername != "" {
+			if opts.nexusPassword == "" {
+				nexusUrl.User = url.User(opts.nexusUsername)
+			} else {
+				nexusUrl.User = url.UserPassword(opts.nexusUsername, opts.nexusPassword)
+			}
+		}
+		nexusAuth := nexusUrl.User.String() // this is encoded as needed.
+		a := strings.SplitN(nexusAuth, ":", 2)
+		un_esc := ""
+		pw_esc := ""
+		if len(a) > 0 {
+			un_esc = a[0]
+		}
+		if len(a) > 1 {
+			pw_esc = a[1]
+		}
+		nexusArgs := []string{
+			fmt.Sprintf("--build-arg=nexusUrl=\"%s\"", opts.nexusURL),
+			fmt.Sprintf("--build-arg=nexusUsername=\"%s\"", un_esc),
+			fmt.Sprintf("--build-arg=nexusPassword=\"%s\"", pw_esc),
+			fmt.Sprintf("--build-arg=nexusHost=\"%s\"", nexusUrl.Host),
+		}
+		args = append(args, nexusArgs...)
```

Comment:
I am slightly confused about this method both appending to `args`, but also returning it ...